internal/slicecache/slicecache.go

```diff
@@ -93,10 +95,17 @@ func (c *FileShardCache) WritethroughReader(shard int, reader sliceio.Reader) sl
 	return newWritethroughReader(reader, c.path(shard))
 }
 
-// CacheReader returns a reader that reads from the cache.
+// CacheReader returns a reader that reads from the cache. If the shard is not
+// cached, returns a reader that will always return an error.
 func (c *FileShardCache) CacheReader(shard int) sliceio.Reader {
 	if !c.shardIsCached[shard] {
-		return sliceio.ErrReader(fmt.Errorf("shard %d is not cached", shard))
+		path := c.path(shard)
+		if c.requireAll {
+			path = fmt.Sprintf("%s-NNNN-of-%04d", c.prefix, c.numShards)
```

Comment:
Optional: Declare this and `path`'s format strings in a top-level block, together, to make sure we don't accidentally change one without the other.